portal/dict_tools.py

```diff
@@ -43,7 +43,7 @@ def dict_match(newd, oldd, diff_stream):
             diff_stream.write(
                 "removed {}\n".format({k:oldd.get(k) for k in removed}))
         if modified:
-            for k in modified.keys():
+            for k in list(modified.keys()):
                 diff_stream.write(
```

Comment:
same as above, we just need to iterate over the keys here - no need to force to be a `list` type